### PR TITLE
Destination Rules

### PR DESCRIPTION
**Pull request by rshriram**

### CODE REVIEW

File: routing/v1alpha2/destination_rule.proto [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+    // balancer will use a random number as the hash, effectively making
+    // the load balancing policy random.
+    string http_header = 1;
+
+    // The minimum number of virtual nodes to use for the hash
+    // ring. Defaults to 1024. Larger ring sizes result in more granular
+    // load distributions. If the number of hosts in the load balancing
+    // pool is larger than the ring size, each host will be assigned a
+    // single virtual node.
+    uint32 minimum_ring_size = 2;
```
Comment (wora) [inline routing/v1alpha2/destination_rule.proto]: All other fields are using int32, why this one is special?

File: routing/v1alpha2/destination_rule.proto [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+//       destination:
+//         name: myredissrv
+//       connectionPool:
+//         tcp:
+//           maxConnections: 100
+//           connectTimeout: 30ms
+//
+message ConnectionPoolSettings {
+  // Settings common to both HTTP and TCP upstream connections.
+  message TCPSettings {
```
Comment (wora) [inline routing/v1alpha2/destination_rule.proto]: Do we use acronyms inside symbols? Envoy and Google APIs don't use acronyms.

File: routing/v1alpha2/destination_rule.proto (hunk 1 above)
Comment (rshriram) [reply routing/v1alpha2/destination_rule.proto]: Copied from Envoy API, which is also using uint32.

File: routing/v1alpha2/destination_rule.proto (hunk 2 above)
Comment (rshriram) [reply routing/v1alpha2/destination_rule.proto]: We have `TCP<*>`, and `HTTP<*>` in several places in our APIs and Envoy APIs.. (https://github.com/envoyproxy/data-plane-api/blob/master/api/filter/http/fault.proto#L26)..

**Pull request merged**